composeApp/src/commonMain/kotlin/sp/bvantur/tasky/core/presentation/SingleEventHandler.kt

```diff
@@ -0,0 +1,25 @@
+package sp.bvantur.tasky.core.presentation
+
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.receiveAsFlow
+import kotlinx.coroutines.withContext
+import sp.bvantur.tasky.core.DispatcherProvider
+
+internal interface SingleEventHandler<Event : SingleEvent> {
+    val singleEventFlow: Flow<Event>
+
+    suspend fun emitSingleEvent(singleEvent: Event)
+}
+
+internal class SingleEventHandlerImpl<Event : SingleEvent>(private val dispatcherProvider: DispatcherProvider) :
+    SingleEventHandler<Event> {
+    private val mutableSingleEventChannel = Channel<Event>(capacity = Channel.BUFFERED)
+    override val singleEventFlow = mutableSingleEventChannel.receiveAsFlow()
+
+    override suspend fun emitSingleEvent(singleEvent: Event) {
+        withContext(dispatcherProvider.main.immediate) {
+            mutableSingleEventChannel.send(singleEvent)
+        }
+    }
+}
```

Comment:
The one and only scenario (that I can spontaneously think of) where you would really make use of this abstraction is in isolated UI tests to simulate sending events. How often do you really have isolated UI tests without testing them together with the ViewModel? (that's in my experience where the test value lies)

Such abstractions always feel very fancy, but if they're not used it's just an extra class that bloats the project. 

PS: Even with the real implementation, you could super easily send events in an isolated UI test because how would the implementation differ there? You'd as well just use a channel.